server.py

```diff
@@ -7,16 +7,21 @@ def main():
     serv.listen(5)
     while True:
         conn, addr = serv.accept()
-        print 'Client from address: {} conn: {}'.format(addr, conn)
+        print 'Client from address: {} '.format(addr)
         data = "Start"
         while data != "":
-            data = conn.recv(4096)
+            try:
+                data = conn.recv(4096)
```

Comment:
What's going to happen if the client calls shutdown without sending "SHUTDOWN", try it & debug the server while you do it

Reply:
I did it before sending you and did it now again. It seems ok, the server prints "Client aborted" and closes the connection.

Reply:
Something else should happen when shutdown is called... try debugging it